suite-native/atoms/src/VStack.tsx

```diff
@@ -1,35 +1,37 @@
 import React, { ReactNode } from 'react';
 
-import { NativeStyleObject, prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
 import { NativeSpacing } from '@trezor/theme';
 
-import { Box } from './Box';
+import { Box, BoxProps } from './Box';
 
-type VStackProps = {
+interface VStackProps extends BoxProps {
     children: ReactNode;
-    style?: NativeStyleObject;
-    spacing: NativeSpacing;
-};
+    spacing?: NativeSpacing | number;
+}
 
 type SpacerStyleProps = {
     isLastChild: boolean;
-    spacing: NativeSpacing;
+    spacing?: NativeSpacing | number;
```

Comment:
This will create more problems than solutions imo opinion.
For example you will need to have some kind of precedence. What if you add both spacing and custom spacing? I would add it to box as well if it's problem that it's not here but that would require more refactoring and could do that within mentioned stack PR. I really don't think we should only restrict spacings to just 4 values from theme (small, medium, ...) if we want to do PX perfect design. Most of the time you will just create additional code for creating styles, applying styles and importing these functions from styles package. We could maybe extend our basic components a little bit so they're more reusable because right now they're not really modifiable apart from margin, padding and flex props.

Reply:
Not sure where link should lead, it leads to project dashboard 😅 

About other components (like `Box` etc.), I don't think we should extend them, it will only create unconsistencies and supports overusing of these props. These props are meant to be used only for quick stylings and you should use like max 3 of them and if you need more you should use normal style prop. Same for additional customization, if you need anything more custom use style prop which is preferred.

When you have prop for style everything you have your render functions full of just style props and styling logic which is not good for readability. 

I don't want to restrict spacing to just for 4 values, but I want to make it to be always preferred if possible. I agree that these two components should be able to accept custom spacing, but I think we should make it "little bit more inconvinient" to do that, which will make you to prefer recommended sizings and that is what `customSpacing` prop should do.

About priority of props you can easily solve this in TS:
```tsx
type SizingProps = {spacing: NativeSpacing} | {customSpacing: number}
type ComponentProps = {
 ...someProps
} & SizingProps
```

Which will allow you to only use one of these prop at time.

Does it makes sense? If not will be better to call about this.

Reply:
Should lead to this issue. When it's open as drawer it doesn't copy the correct link, sorry. 


https://github.com/trezor/trezor-suite/issues/5704